libclc/libspirv/lib/amdgcn-amdhsa/group/collectives.cl

```diff
@@ -38,24 +38,23 @@ __clc__get_group_scratch_float() __asm("__clc__get_group_scratch_float");
 __local double *
 __clc__get_group_scratch_double() __asm("__clc__get_group_scratch_double");
 
-#define __CLC_DECLARE_SHUFFLES(TYPE, TYPE_MANGLED)                             \
-  _CLC_DECL TYPE _Z28__spirv_SubgroupShuffleINTELI##TYPE_MANGLED##ET_S0_j(     \
-      TYPE, int);                                                              \
-  _CLC_DECL TYPE                                                               \
-      _Z30__spirv_SubgroupShuffleUpINTELI##TYPE_MANGLED##ET_S0_S0_j(           \
-          TYPE, TYPE, unsigned int);
-
-__CLC_DECLARE_SHUFFLES(char, a);
-__CLC_DECLARE_SHUFFLES(unsigned char, h);
-__CLC_DECLARE_SHUFFLES(short, s);
-__CLC_DECLARE_SHUFFLES(unsigned short, t);
-__CLC_DECLARE_SHUFFLES(int, i);
-__CLC_DECLARE_SHUFFLES(unsigned int, j);
-__CLC_DECLARE_SHUFFLES(half, DF16_);
-__CLC_DECLARE_SHUFFLES(float, f);
-__CLC_DECLARE_SHUFFLES(long, l);
-__CLC_DECLARE_SHUFFLES(unsigned long, m);
-__CLC_DECLARE_SHUFFLES(double, d);
+#define __CLC_DECLARE_SHUFFLES(TYPE)                             \
+  _CLC_OVERLOAD _CLC_DECL TYPE __spirv_SubgroupShuffleINTEL(     \
```

Comment:
I'm not sure if we can remove implementation of existing built-ins. Won't this be an ABI breaking change? The main question is: can we link static libraries produced by the old compiler using new compiler?

Reply:
Yes it would be an ABI breakage for device code and we can't link link static libraries produced by the old compiler using new compiler.
Should I keep all the existing implementation?
We probably need to add note for either removal or deprecation in upcoming release notes.

Reply:
> Should I keep all the existing implementation?

Yes. We should keep old implementation to maintain stable ABI as required by the [Contribution guidelines](https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#maintaining-stable-abiapi).

I'm not sure about release notes. This is implementation detail, which is not visible to users. We are allowed to break ABI with major version changes. 

@AlexeySachkov, FYI.

Reply:
added back old implementations.